api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirConditionDaoImpl.java

```diff
@@ -0,0 +1,135 @@
+import org.openmrs.Obs;
+import org.openmrs.annotation.Authorized;
+import org.openmrs.annotation.OpenmrsProfile;
+import org.openmrs.module.fhir2.FhirConstants;
+import org.openmrs.module.fhir2.api.dao.FhirConditionDao;
+import org.openmrs.module.fhir2.api.search.param.SearchParameterMap;
+import org.openmrs.util.PrivilegeConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+@Component
+@Setter(AccessLevel.PUBLIC)
+@OpenmrsProfile(openmrsPlatformVersion = "2.0.5 - 2.1.*")
+public class FhirConditionDaoImpl extends BaseFhirDao<Obs> implements FhirConditionDao<Obs> {
+	
+	@Qualifier("sessionFactory")
+	@Autowired
+	private SessionFactory sessionFactory;
+	
+	@Override
+	@Authorized(PrivilegeConstants.GET_OBS)
+	public Obs get(@Nonnull String uuid) {
+		return super.get(uuid);
+	}
+	
+	@Override
+	@Authorized(PrivilegeConstants.EDIT_OBS)
+	public Obs createOrUpdate(@Nonnull Obs newEntry) {
+		return super.createOrUpdate(newEntry);
+	}
+	
+	@Override
+	@Authorized(PrivilegeConstants.DELETE_OBS)
+	public Obs delete(@Nonnull String uuid) {
+		return super.delete(uuid);
+	}
+	
+	@Override
+	@Authorized(PrivilegeConstants.GET_OBS)
+	public List<String> getSearchResultUuids(@Nonnull SearchParameterMap theParams) {
+		return super.getSearchResultUuids(theParams);
+	}
+	
+	@Override
+	@Authorized(PrivilegeConstants.GET_OBS)
+	public List<Obs> getSearchResults(@Nonnull SearchParameterMap theParams, @Nonnull List<String> matchingResourceUuids,
+	        int firstResult, int lastResult) {
+		return super.getSearchResults(theParams, matchingResourceUuids, firstResult, lastResult);
+	}
+	
+	@Override
+	protected void setupSearchParams(Criteria criteria, SearchParameterMap theParams) {
+		theParams.getParameters().forEach(entry -> {
+			switch (entry.getKey()) {
+				case FhirConstants.PATIENT_REFERENCE_SEARCH_HANDLER:
+					entry.getValue().forEach(
+					    param -> handlePatientReference(criteria, (ReferenceAndListParam) param.getParam(), "person"));
+					break;
+				case FhirConstants.CODED_SEARCH_HANDLER:
+					entry.getValue().forEach(param -> handleCode(criteria, (TokenAndListParam) param.getParam()));
+					break;
+				case FhirConstants.DATE_RANGE_SEARCH_HANDLER:
+					entry.getValue()
+					        .forEach(param -> handleDateRange(param.getPropertyName(), (DateRangeParam) param.getParam())
+					                .ifPresent(criteria::add));
+					break;
+				case FhirConstants.QUANTITY_SEARCH_HANDLER:
+					entry.getValue().forEach(param -> handleOnsetAge(criteria, (QuantityAndListParam) param.getParam()));
+					break;
+				case FhirConstants.COMMON_SEARCH_HANDLER:
+					handleCommonSearchParameters(entry.getValue()).ifPresent(criteria::add);
+					break;
+			}
+		});
+		criteria.createAlias("concept", "conc");
+		criteria.add(eq("conc.uuid", FhirConstants.CONDITION_OBSERVATION_CONCEPT_UUID));
```

Comment:
I'd prefer this to be aliased to `"c"` for consistency. Also, this should go before the `forEach` block.